x/ecocredit/client/testsuite/tx.go

```diff
@@ -279,162 +119,66 @@ func makeFlagFrom(from string) string {
 
 func (s *IntegrationTestSuite) TestTxCreateClass() {
 	val0 := s.network.Validators[0]
-	val1 := s.network.Validators[1]
 	clientCtx := val0.ClientCtx
+	fee := core.DefaultParams().CreditClassFee[0]
+	feeStr := fee.String()
 
 	testCases := []struct {
 		name              string
 		args              []string
 		expectErr         bool
 		expectedErrMsg    string
 		respCode          uint32
-		expectedClassInfo *ecocredit.ClassInfo
+		expectedClassInfo *core.ClassInfo
 	}{
 		{
 			name:           "missing args",
 			args:           []string{},
 			expectErr:      true,
-			expectedErrMsg: "accepts 3 arg(s), received 0",
+			expectedErrMsg: "accepts 4 arg(s), received 0",
 		},
 		{
 			name:           "too many args",
-			args:           []string{"abcde", "abcde", "abcde", "abcde"},
-			expectErr:      true,
-			expectedErrMsg: "accepts 3 arg(s), received 4",
-		},
-		{
-			name: "invalid issuer",
-			args: append(
-				[]string{
-					"abcde",
-					validCreditType,
-					validMetadata,
-					makeFlagFrom(val0.Address.String()),
-				},
-				s.commonTxFlags()...,
-			),
+			args:           []string{"abcde", "abcde", "abcde", "abcde", "abce"},
 			expectErr:      true,
-			expectedErrMsg: "decoding bech32 failed: invalid bech32 string length 5",
+			expectedErrMsg: "accepts 4 arg(s), received 5",
 		},
 		{
-			name: "invalid metadata",
-			args: append(
-				[]string{
-					val0.Address.String(),
-					validCreditType,
-					"=",
-					makeFlagFrom(val0.Address.String()),
-				},
-				s.commonTxFlags()...,
-			),
-			expectErr:      true,
-			expectedErrMsg: "metadata is malformed, proper base64 string is required",
-		},
-		{
-			name: "missing from flag",
-			args: append(
-				[]string{
-					val0.Address.String(),
-					validCreditType,
-					validMetadata,
-				},
-				s.commonTxFlags()...,
-			),
+			name:           "missing from flag",
+			args:           makeCreateClassArgs([]string{val0.Address.String()}, validCreditTypeAbbrev, validMetadata, feeStr, s.commonTxFlags()...),
 			expectErr:      true,
 			expectedErrMsg: "required flag(s) \"from\" not set",
 		},
 		{
-			name: "invalid credit type",
-			args: append(
-				[]string{
-					val0.Address.String(),
-					"caarbon",
-					validMetadata,
-					makeFlagFrom(val0.Address.String()),
-				},
-				s.commonTxFlags()...,
-			),
-			expectErr:      false,
-			expectedErrMsg: "caarbon is not a valid credit type",
-			respCode:       29,
-		},
-		{
-			name: "single issuer",
-			args: append(
-				[]string{
-					val0.Address.String(),
-					validCreditType,
-					validMetadata,
-					makeFlagFrom(val0.Address.String()),
-				},
-				s.commonTxFlags()...,
-			),
-			expectErr: false,
-			expectedClassInfo: &ecocredit.ClassInfo{
-				Admin:    val0.Address.String(),
-				Issuers:  []string{val0.Address.String()},
-				Metadata: []byte{0x1},
-			},
-		},
```

Comment:
All of these are being removed here and in other client tests because they are already tested with unit tests...? I think this is ok but just want to make sure we are double checking.

Reply:
yes, request data validity seems more of a unit-level than integraiton level concern